src/server/action_process_queue.ts

```diff
@@ -0,0 +1,29 @@
+import * as spawn from "child_process"
+import * as winston from "winston"
+import {ProcessQueue} from "./process_queue"
+
+export class ExecuteProcessQueue extends ProcessQueue {
+
+    async run(data: string) {
+        return this.queue.add(async () => {
+            return new Promise<string>((resolve, reject) => {
+                const child = spawn.fork(`./src/actions/actions_process.ts`)
```

Comment:
can we call this `execute_process` instead for clarity?

Reply:
I changed it to execute_process_queue . I personally like that name more

Reply:
actually, its weird that `ExecuteProcessQueue` is in `src/server/action_process_queue.ts` and then there's also `execute_process_queue` now?

Maybe 

- `ExecuteProcessQueue` is in `src/xpc/execute_process_queue.ts` (in `xpc` folder instead of `server`)
- the actual "worker" in in `src/xpc/execute_worker.ts`